io/src/main/scala/fs2/io/io.scala

```diff
@@ -178,4 +178,16 @@ package object io {
   def toInputStreamResource[F[_]](source: Stream[F, Byte])(
       implicit F: ConcurrentEffect[F]): Resource[F, InputStream] =
     JavaInputOutputStream.toInputStream(source)
+
+  /**
+    * Like `Sync#delay` but evaluates the thunk on the supplied execution context
+    * and then shifts back to the default thread pool.
+    */
+  private[io] def blockingDelay[F[_], A](blockingExecutionContext: ExecutionContext)(
+      thunk: => A)(implicit F: Sync[F], cs: ContextShift[F]): F[A] =
+    cs.evalOn(blockingExecutionContext)(F.delay(blocking(thunk)))
```

Comment:
Is the use of `blocking` needed here? I was under the impression that pretty much only `global` deals with it, and if the cats-effect change goes through, we'd probably always use a `cachedThreadPool` here anyway

Reply:
Yeah, only used with contexts backed by FJPool. I'll remove

Reply:
The `blockingExecutionContext` seem to be passed in through a public API. Which means, that it may be a ForkJoinPool (even if it's not advisable). In which case `blocking` is useful. And in other cases it doesn't hurt. (Or does it?)

Reply:
The idea is that it will be replaced by https://github.com/typelevel/cats-effect/pull/556

Reply:
I agree that the `blocking` call doesn't hurt though I don't want to encourage use of FJPool as that relies on all blocking thunks to be wrapped in `blocking`, which isn't true in the wider cats-effect ecosystem.

Reply:
Ok, I didn't realize the `ExecutionContext` will be replaced by this new `Blocker`.